src/core/main.c

```diff
@@ -354,11 +384,9 @@ static int parse_proc_cmdline_item(const char *key, const char *value) {
 
         } else if (streq(key, "systemd.confirm_spawn") && value) {
 
-                r = parse_boolean(value);
+                r = parse_confirm_spawn(value, &arg_confirm_spawn);
                 if (r < 0)
-                        log_warning("Failed to parse confirm spawn switch %s. Ignoring.", value);
-                else
-                        arg_confirm_spawn = r;
+                        log_warning("Failed to parse confirm_spawn switch %s. Ignoring.", value);
```

Comment:
while you are at it, this should really be changed to `log_warning_errno(r,…` here... If we have an error number we should pass it to the logging function.